paasta_tools/setup_marathon_job.py

```diff
@@ -554,13 +557,13 @@ def log_deploy_error(errormsg, level='event'):
     except bounce_lib.LockHeldException:
         logline = 'Failed to get lock to create marathon app for %s.%s' % (service, instance)
         log_deploy_error(logline, level='debug')
-        return (0, "Couldn't get marathon lock, skipping until next time")
```

Comment:
Are you afraid of an infinite loop here? If we can't get a lock, then shouldn't we return -1 because another one beat us to it?